Tests/Marketplace/marketplace_services.py

```diff
@@ -1233,6 +1235,91 @@ def get_changelog_latest_rn(self, changelog_index_path: str) -> Tuple[dict, Loos
+            A dict of modified version and their release notes contents, for modified
+              in the current index file
+
+
+        """
+        # No modified rn files, return None
+        if len(modified_rn_files) == 0:
+            return None
+
+        modified_versions_dict = {}
+
+        # Found modified rn files
+        logging.info("Found modified rn files")
+        for rn_filename in modified_rn_files:
+            _version = rn_filename.replace('.md', '')
+            version = _version.replace('_', '.')
+            # Should only apply on modified files that are not the last rn file
+            if LooseVersion(version) < changelog_latest_rn_version:
+                # The case where the version is a key in the changelog file,
+                # and the value is not an aggregated release note
+                if is_the_only_rn_in_block(release_notes_dir, version, changelog):
+                    logging.info("The version is a key in the changelog file and alone")
+                    with open(os.path.join(release_notes_dir, rn_filename), 'r') as rn_file:
+                        rn_lines = rn_file.read()
+                    modified_versions_dict[version] = self._clean_release_notes(rn_lines).strip()
+                # The case where the version is not a key in the changelog file or it is a key of aggregated content
+                else:
+                    logging.info("The version is not a key in the changelog file or it is a key of aggregated content")
+                    same_block_versions_dict, higher_nearest_version = self.get_same_block_versions(
+                        release_notes_dir, version, changelog)
+                    modified_versions_dict[higher_nearest_version] = aggregate_release_notes_for_marketplace(
+                        same_block_versions_dict)
+
+        return modified_versions_dict
+
+    def get_same_block_versions(self, release_notes_dir: str, version: str, changelog: dict):
+        """
+        Get a dict of the version as key and rn data as value of all of the versions that are in the same
+        block in the changelog file as the given version (these are the versions that were aggregates together
+        during a single upload priorly).
+
+        Args:
+            release_notes_dir (str): the path to the release notes dir
+            version (str): the wanted version
+            changelog (dict): the changelog from the production bucket.
+
+        Returns:
+            A dict of version, rn data for all corresponding versions, and the highest version among those keys as str
+
+        """
+        same_block_versions_dict: dict = dict()
+        versions = [LooseVersion(item) for item in changelog.keys()]
+        higher_versions = [item for item in versions if item > version]
+        lower_versions = [item for item in versions if item < version]
+        higher_nearest_version = min(higher_versions)
+        lower_nearest_version = max(lower_versions)
```

Comment:
This code is iterating on `changelog-keys` 5 times, better to do it in one iteration.